implementations/rust/ockam/ockam_api/src/cloud/mod.rs

```diff
@@ -42,6 +42,7 @@ pub struct ProjectUserRole {
     #[n(1)] pub email: String,
     #[n(2)] pub id: usize,
     #[n(3)] pub role: RoleInShare,
+    #[n(4)] pub scope: ShareScope,
```

Comment:
I don't see this field being used anywhere in the Rust code. Is it supposed to be here?

Reply:
> What is it used for? Is it set by the Orchestrator?

That's correct, and the general purpose is the same as when it was introduced during #5451. Users may hold non-overlapping roles in a Project and the Space that Project belongs to, so this can be used for client-side logic to distinguish when needed. For example, a user can be a Guest in a Space and an Admin in a Project within that Space. In that kind of combination, without this disambiguating field, the `user_roles` list would appear like:
```json
[
  {"email": "user@example.com", "id": 12345, "role": "Guest" },
  {"email": "user@example.com", "id": 12345, "role": "Admin" }
]
```
After:
```json
[
  {"email": "user@example.com", "id": 12345, "role": "Guest", "scope": "Space" },
  {"email": "user@example.com", "id": 12345, "role": "Admin", "scope": "Project" }
]
```

Reply:
Ok. Given the name `ProjectUserRole` I was expecting that the `scope` was already `Project` and that the query for getting the `user_roles` was already specifying for which project. Isn't there a bit of mismatch in names then?

Reply:
> Given the name ProjectUserRole I was expecting that the scope was already Project and that the query for getting the user_roles was already specifying for which project.

Perhaps renaming `scope` to something reflecting "origin/source of this granted role" would've reduced or avoided your partially-incorrect assumption here. The `user_roles` list is indeed specific to the individual project that appears in. Read on if you want more color.

---

Today, most users with access to a Project, especially in the relatively common case where they are newly enrolled are accessing the Project as an Admin of the Space the project belongs to rather than having direct rights attached to the Project itself. When more specific information is not present, a user's effective Project rights are inherited from their rights on the parent Space.

The backend previously did not report the access levels that are conveyed from the parent Space in this response payload, only the ones that are directly attached to the Project. Taken together with the two points above, this meant that the `user_roles` list was nearly always empty (and thus unhelpful) in the majority of scenarios. This was recently changed on the backend for more complete information.

The reason any of this matters is because the existing client behavior often relies far too much on underspecified inference, which was often developed without input or feedback from team members who are intimately familiar with the backend implementation details.

TLDR - When creating invitations, or anywhere else we'll need to display candidates for a user to take similar elevated actions on, we actually only care about identifying projects where the user is an acting Admin, regardless of where those rights are specifically granted.